### PR TITLE
directory for metadata cache

### LINKED ISSUE #64: stop piling up of XML files with file links?
XML files with download links that are created once a day whenever `spod_get` or almost any other function is used are piling up in the `SPANISH_OD_DATA_DIR`. We need a good mechanism to get rid of them... On one hand, there are not that many of them and compared to the data itself, they are not taking up space, but do we really need old XML files...? We need a good solution.

One thing I am considering, is to at least create a dedicated folder for them, instead of dumping them in the root of `SPANISH_OD_DATA_DIR`.

@Robinlovelace suggestions? (not urgent)

### PR DESCRIPTION
**Pull request by e-kotov**
Fixes #64 .

### CODE REVIEW

File: R/folders.R [hunk 1]
```diff
@@ -1,3 +1,12 @@
+#' Get metadata cache subfolder name
+#'
+#' Change subfolder name in the code of this function for metadata cache here to apply globally, as all functions in the package should use this function to get the metadata cache path.
+#' @return Character string with the subfolder name for the raw data cache.
+#' @keywords internal
+spod_subfolder_metadata_cache <- function() {
```
Comment (Robinlovelace) [inline R/folders.R:6]: Do we we really need a function that just returns a single text string? Not sure how `spod_subfolder_metadata_cache()` could be better than "metadata_cache" but imagine there's a reason..

**Review by Robinlovelace — APPROVED** [review]
Good to keep the metadata cache in a subfolder.

File: R/folders.R (hunk 1 above)
Comment (e-kotov) [reply R/folders.R:6]: maybe it is a bit of over engineering, but at least we will only have to change it in one place if we want to change it. better solutions? package option?

Comment (Robinlovelace) [reply R/folders.R:6]: No better solutions, just checking the logic and fine by me for this to go into the pkg.

**Pull request merged**